test/compile-error-test/CMakeLists.txt.in

```diff
@@ -0,0 +1,6 @@
+cmake_minimum_required(VERSION 3.1...3.18)
+project(tests)
```

Comment:
This is probably not needed.

Reply:
We are creating this `tests` tree as a separate project that builds outside of the {fmt} CMake project by calling CMake directly on configured CMakeLists.txt file. This makes it top-level CMakeLists.txt, and according to [CMake docs](https://cmake.org/cmake/help/latest/command/project.html#usage) (and my eyes), we should add a `project` call to avoid getting the warning described in docs.